include/exadg/acoustic_conservation_equations/spatial_discretization/spatial_operator.cpp

```diff
@@ -290,29 +288,16 @@ void
 SpatialOperator<dim, Number>::prescribe_initial_conditions(BlockVectorType & dst,
                                                            double const      time) const
 {
-  field_functions->initial_solution_pressure->set_time(time);
-  field_functions->initial_solution_velocity->set_time(time);
-
-  // This is necessary if Number == float
-  using VectorTypeDouble = dealii::LinearAlgebra::distributed::Vector<double>;
-
-  VectorTypeDouble pressure_double;
-  VectorTypeDouble velocity_double;
-  pressure_double = dst.block(block_index_pressure);
-  velocity_double = dst.block(block_index_velocity);
-
-  dealii::VectorTools::interpolate(*get_mapping(),
-                                   dof_handler_p,
-                                   *(field_functions->initial_solution_pressure),
-                                   pressure_double);
-
-  dealii::VectorTools::interpolate(*get_mapping(),
-                                   dof_handler_u,
-                                   *(field_functions->initial_solution_velocity),
-                                   velocity_double);
-
-  dst.block(block_index_pressure) = pressure_double;
-  dst.block(block_index_velocity) = velocity_double;
+  Utilities::interpolate(*get_mapping(),
+                         dof_handler_p,
+                         *(field_functions->initial_solution_pressure),
+                         dst.block(block_index_pressure),
+                         time);
+  Utilities::interpolate(*get_mapping(),
+                         dof_handler_u,
+                         *(field_functions->initial_solution_velocity),
+                         dst.block(block_index_velocity),
+                         time);
```

Comment:
Did you think about using the member function `interpolate_functions()` (for both velocity and pressure) here, or is it not applicable here? Using that function might give a more consistent picture overall, but maybe I overlooked something.

Reply:
The member function `interpolate_functions()` is defined in the IncNS operator and uses the fluid mapping and dof handlers . Since we do not have multiple interpolation functions in this operator I would keep it this way :)